mypy/messages.py

```diff
@@ -136,8 +137,11 @@ def add_errors(self, messages: 'MessageBuilder') -> None:
                 for info in errs:
                     self.errors.add_error_info(info)
 
-    def disable_errors(self) -> None:
+    @contextmanager
+    def disable_errors(self) -> Iterator[None]:
         self.disable_count += 1
+        yield
```

Comment:
Should use try-finally

Reply:
Thank you for the review @JelleZijlstra 

yield should be inside try-finally? 

`@contextmanager
    def disable_errors(self) -> Iterator[None]:
        self.disable_count += 1
        try:
            yield
        finally:
            pass
        self.disable_count -= 1`

Reply:
And the `-= 1` should be in the finally block. The point is to make sure the cleanup code always happens.

Reply:
Got it.

Reply:
@JelleZijlstra Now that we have try-finally inside `enable_errors()` below try-finally is redundent right? Can be removed? https://github.com/python/mypy/blob/8860575a2134d87d890119015a9ead172f7fdfb1/mypy/checkexpr.py#L2811

```python
if right_map is None:
  with self.msg.disable_errors():
     try:
        right_type = self.analyze_cond_branch(right_map, e.right, left_type)
     finally:
        pass
```

Reply:
A try-finally with an empty finally block doesn't do anything useful, so yes you can remove it.